### ISSUE TITLE
v4.1.x: Compile fail on --enable-heterogeneous builds

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#9861]
[From Cisco MTT](https://mtt.open-mpi.org/index.php?limit=&wrap=&trial=&enable_drilldowns=&yaxis_scale=&xaxis_scale=&hide_subtitle=&split_graphs=&remote_go=&do_cookies=&phase=mpi_install&text_start_timestamp=2022-01-10+16%3A30%3A01+-+2022-01-11+16%3A30%3A01&text_platform_hardware=%5Ex86_64%24&show_platform_hardware=show&text_os_name=%5ELinux%24&show_os_name=show&text_mpi_name=%5Eompi-nightly-v4.1.x%24&show_mpi_name=show&text_mpi_version=%5Ev4.1.x-202201110400-5ef48ea%24&show_mpi_version=show&text_configure_arguments=&text_compiler_name=&show_compiler_name=show&text_bitness=&show_bitness=show&text_endian=&show_endian=show&text_http_username=%5Ecisco%24&show_http_username=show&text_local_username=all&show_local_username=hide&text_platform_name=%5Ecisco-community%2C+usNIC+build%3Dmtt-usnic-890.0%24&show_platform_name=show&click=Detail&phase=mpi_install&test_result=_mpi_f&text_compute_cluster_id=&text_os_version=&show_os_version=&text_platform_type=&show_platform_type=&text_submit_id=&text_hostname=&show_hostname=&text_mpi_get_id=&text_mpi_install_compiler_id=&text_mpi_install_configure_id=&text_mpi_install_id=&text_compiler_version=&show_compiler_version=&text_vpath_mode=&show_vpath_mode=&text_exit_value=&show_exit_value=&text_exit_signal=&show_exit_signal=&text_duration=&show_duration=&text_client_serial=&show_client_serial=&text_result_message=&text_result_stdout=&text_result_stderr=&text_environment=&text_description=&text_trial=%5Ef%24&lastgo=summary):

```
make[1]: Entering directory `/home/jsquyres/git/o4/ompi/mca/coll'
  CC       base/coll_base_alltoall.lo
In file included from ../../../opal/threads/thread_usage.h:31,
                 from ../../../opal/class/opal_object.h:126,
                 from ../../../opal/datatype/opal_datatype.h:41,
                 from ../../../opal/datatype/opal_convertor.h:35,
                 from ../../../ompi/datatype/ompi_datatype.h:38,
                 from base/coll_base_alltoall.c:31:
base/coll_base_alltoall.c: In function ‘mca_coll_base_alltoall_intra_basic_inplace’:
base/coll_base_alltoall.c:81:85: error: dereferencing pointer to incomplete type ‘struct opal_convertor_master_t’
 if( OPAL_UNLIKELY(opal_local_arch != ompi_proc->super.proc_convertor->master->remote_arch))  {
                                                                             ^~

../../../opal/include/opal/prefetch.h:44:55: note: in definition of macro ‘OPAL_UNLIKEL’
 #define OPAL_UNLIKELY(expression) __builtin_expect(!!(expression), 0)
                                                       ^~~~~~~~~~
make[1]: *** [base/coll_base_alltoall.lo] Error 1
make[1]: Leaving directory `/home/jsquyres/git/o4/ompi/mca/coll'
make: *** [all-recursive] Error 1
```

**Comment by bosilca** [comment]
Missing 927e9aa97373dac652f9cba4813e6ee609ca2830.

**Comment by jsquyres** [comment]
This was fixed by #9863
